src/components/InventoryTable/utils/api.js

```diff
@@ -0,0 +1,25 @@
+import pAll from 'p-all';
+import { hosts } from '../../../api';
+
+const resolve = async (fns, limit = 2) => {
+    const results = await pAll(fns, {
+        concurrency: limit
+    });
+    return results;
+};
+
+export const deleteSystemsById = (items, batchSize = 50) => {
+    let arr = [];
+    for (let i = 0; i < items.length; i += batchSize) {
+        let chunk;
+        chunk = items.slice(i, i + batchSize);
+        arr.push(chunk);
+    }
+
+    const results = resolve(
+        arr.map(itemArray => () => {
+            return hosts.apiHostDeleteById(itemArray); }
```

Comment:
We need to have https://github.com/RedHatInsights/insights-inventory-frontend/pull/1858 merged first, the naming for these API functions have changed